python/pyarrow/_azurefs.pyx

```diff
@@ -107,7 +128,32 @@ cdef class AzureFileSystem(FileSystem):
         if account_key and sas_token:
             raise ValueError("Cannot specify both account_key and sas_token.")
 
-        if account_key:
+        if (tenant_id or client_id or client_secret):
+            if tenant_id and client_id and client_secret:
```

Comment:
I think the previous iteration was clearer:
- If there's no client_id fail early without requiring more checks
- If there's no tenant and no client_secret --> ConfigureManagedIdentityCredential
- If there's tenant and client_secret --> ConfigureClientSecretCredential
- Otherwise fail